server/src/test/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponseTests.java

```diff
@@ -0,0 +1,191 @@
+            }
+        }
+
+        ImmutableOpenMap<String, Settings> immutableIndexToDefaultSettings =
+            ImmutableOpenMap.<String, Settings>builder().putAll(indexToDefaultSettings).build();
+
+        return new GetSettingsResponse(immutableIndexToSettings, immutableIndexToDefaultSettings);
+    }
+
+    @Override
+    protected GetSettingsResponse doParseInstance(XContentParser parser) throws IOException {
+        return GetSettingsResponse.fromXContent(parser);
+    }
+
+    protected boolean supportsUnknownFields() {
+        return false;
+    }
+
+    private static GetSettingsResponse getExpectedTest622Response() {
+    /* This is a fairly direct copy of the code used to generate the base64'd response above -- with the caveat that the constructor
+    has been modified so that the code compiles on this version of elasticsearch
+     */
+        HashMap<String, Settings> indexToSettings = new HashMap<>();
+        Settings.Builder builder = Settings.builder();
+
+        builder.put(SETTING_NUMBER_OF_SHARDS, 2);
+        builder.put(SETTING_NUMBER_OF_REPLICAS, 1);
+        indexToSettings.put("index_name", builder.build());
+        GetSettingsResponse response = new GetSettingsResponse(ImmutableOpenMap.<String, Settings>builder().putAll(indexToSettings).build
+            (), ImmutableOpenMap.of());
+        return response;
+    }
+
+    private static GetSettingsResponse getResponseWithNewFields() {
+        HashMap<String, Settings> indexToDefaultSettings = new HashMap<>();
+        Settings.Builder builder = Settings.builder();
+
+        builder.put(INDEX_REFRESH_INTERVAL_SETTING.getKey(), "1s");
+        indexToDefaultSettings.put("index_name", builder.build());
+        ImmutableOpenMap<String, Settings> defaultsMap = ImmutableOpenMap.<String, Settings>builder().putAll(indexToDefaultSettings)
+            .build();
+        return new GetSettingsResponse(getExpectedTest622Response().getIndexToSettings(), defaultsMap);
+    }
+
+    public void testCanDecode622Response() throws IOException {
+        StreamInput si = StreamInput.wrap(Base64.getDecoder().decode(TEST_6_2_2_RESPONSE_BYTES));
+        si.setVersion(Version.V_6_2_2);
+        GetSettingsResponse response = new GetSettingsResponse();
+        response.readFrom(si);
+
+        Assert.assertEquals(TEST_6_2_2_RESPONSE_INSTANCE, response);
+    }
+
+    public void testCanOutput622Response() throws IOException {
+        GetSettingsResponse responseWithExtraFields = getResponseWithNewFields();
+        BytesStreamOutput bso = new BytesStreamOutput();
+        bso.setVersion(Version.V_6_2_2);
+        responseWithExtraFields.writeTo(bso);
+
+        String base64OfResponse = Base64.getEncoder().encodeToString(BytesReference.toBytes(bso.bytes()));
+
+        Assert.assertEquals(TEST_6_2_2_RESPONSE_BYTES, base64OfResponse);
+    }
+
+    public void testTransportSerdeRoundTripCurrentVersion() throws IOException {
```

Comment:
TransportSerde? what does the suffix mean?

Reply:
Serialization/Deserialization